e2e/testsuite/testsuite.go

```diff
@@ -48,20 +48,58 @@ type E2ETestSuite struct {
 	testifysuite.Suite
 
 	// proposalIDs keeps track of the active proposal ID for each chain.
-	proposalIDs    map[string]uint64
-	paths          map[string][]ibc.Chain
+	proposalIDs map[string]uint64
+
+	// chains is a list of chains that are created for the test suite.
+	// each test suite has a single slice of chains that are used for all individual test
+	// cases.
+	chains         []ibc.Chain
 	relayers       relayer.Map
 	logger         *zap.Logger
 	DockerClient   *dockerclient.Client
 	network        string
 	startRelayerFn func(relayer ibc.Relayer)
 
-	// pathNameIndex is the latest index to be used for generating paths
+	// pathNameIndex is the latest index to be used for generating chains
 	pathNameIndex int64
+
+	// TODO: refactor this to use a relayer per test
+	// relayer is a single relayer which only works when running tests one per host.
+	// this needs to be refactored to use a different relayer per test.
+	relayer ibc.Relayer
+
+	// testSuiteName is the name of the test suite, used to store chains under the test suite name.
+	testSuiteName string
+	testPaths     map[string][]string
+	channels      map[string]map[ibc.Chain][]ibc.ChannelOutput
 }
 
-func (s *E2ETestSuite) SetupTest() {
-	s.configureGenesisDebugExport()
+// initState populates variables that are used across the test suite.
+// note: this should be called only from SetupSuite.
+func (s *E2ETestSuite) initState() {
+	s.initDockerClient()
+	s.proposalIDs = map[string]uint64{}
+	s.testPaths = make(map[string][]string)
+	s.channels = make(map[string]map[ibc.Chain][]ibc.ChannelOutput)
+
+	// testSuiteName gets populated in the context of SetupSuite and stored as s.T().Name()
+	// will return the name of the suite and test when called from SetupTest or within the body of tests.
+	// the chains will be stored under the test suite name, so we need to store this for future use.
+	s.testSuiteName = s.T().Name()
+}
+
+// initDockerClient creates a docker client and populates the network to be used for the test.
+func (s *E2ETestSuite) initDockerClient() {
+	client, network := interchaintest.DockerSetup(s.T())
+	s.logger = zap.NewExample()
+	s.DockerClient = client
+	s.network = network
+}
+
+// SetupSuite will by default create chains with no additional options. If additional options are required,
+// the test suite must define the SetupSuite function and provide the required options.
+func (s *E2ETestSuite) SetupSuite() {
```

Comment:
Just to be clear, this means that all tests in one suite will share the same chains, and so we must be careful that accounts are not shared across the tests to avoid account sequence mismatches, resusing balances, etc, etc.

Reply:
yes that's correct! By default we are always generating addresses that won't collide by default, but we need to keep this in mind in case we end up encountering a new category of error

Reply:
Yeah, I think that will help a lot with running e2e tests faster, which is a big win!